pkg/resolution/resolver/git/resolver.go

```diff
@@ -180,6 +182,33 @@ func ResolveAnonymousGit(ctx context.Context, params map[string]string) (framewo
 	cloneOpts := &git.CloneOptions{
 		URL: repo,
 	}
+
+	secretRef := &secretCacheKey{
+		name: params[GitTokenParam],
+		key:  params[GitTokenKeyParam],
+	}
+	if secretRef.name != "" {
+		if secretRef.key == "" {
+			secretRef.key = DefaultTokenKeyParam
+		}
+		secretRef.ns = common.RequestNamespace(ctx)
+	} else {
+		secretRef = nil
+	}
+
+	auth := plumbTransport.AuthMethod(nil)
+	if secretRef != nil {
+		gitToken, err := getAPIToken(ctx, secretRef, kubeclient, logger, cache, ttl, params, GitTokenKeyParam)
+		if err != nil {
+			return nil, err
+		}
+		auth = &http.BasicAuth{
+			Username: "git",
+			Password: string(gitToken),
+		}
```

Comment:
nit: inline error handling

Reply:
we need access to gitToken after that which would be redudant to declare the variable priors

Reply:
got it